src/utils/updater.py

```
@@ -19,16 +22,61 @@
+    # A source checkout may intentionally be ahead of the latest release tag.
+    # Its project metadata is therefore the source of truth.
+    try:
+        toml_path = PROJECT_ROOT / "pyproject.toml"
+        if toml_path.exists():
+            content = toml_path.read_text(encoding="utf-8")
+            for line in content.splitlines():
+                if line.strip().startswith("version"):
+                    # version = "2.2.0"
+                    return line.split("=")[1].strip().strip('"').strip("'")
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)